src/Runner.Common/JobServerQueue.cs

```diff
@@ -109,25 +111,34 @@ public void Start(Pipelines.AgentJobRequestMessage jobRequest, bool resultServic
             }
 
             // This code is usually wrapped by an instance of IExecutionContext which isn't available here.
-            jobRequest.Variables.TryGetValue("system.github.results_endpoint", out VariableValue resultsEndpointVariable);
-            var resultsReceiverEndpoint = resultsEndpointVariable?.Value;
-
             if (serviceEndPoint?.Authorization != null &&
                 serviceEndPoint.Authorization.Parameters.TryGetValue("AccessToken", out var accessToken) &&
-                !string.IsNullOrEmpty(accessToken) &&
-                !string.IsNullOrEmpty(resultsReceiverEndpoint))
+                !string.IsNullOrEmpty(accessToken))
             {
-                string liveConsoleFeedUrl = null;
-                Trace.Info("Initializing results client");
-                if (resultServiceOnly
+                jobRequest.Variables.TryGetValue("system.github.results_endpoint", out VariableValue resultsEndpointVariable);
+                var resultsReceiverEndpoint = resultsEndpointVariable?.Value;
+                if (!string.IsNullOrEmpty(resultsReceiverEndpoint))
+                {
+                    string liveConsoleFeedUrl = null;
+                    Trace.Info("Initializing results client");
+                    if (resultServiceOnly
                     && serviceEndPoint.Data.TryGetValue("FeedStreamUrl", out var feedStreamUrl)
                     && !string.IsNullOrEmpty(feedStreamUrl))
-                {
-                    liveConsoleFeedUrl = feedStreamUrl;
+                    {
+                        liveConsoleFeedUrl = feedStreamUrl;
+                    }
+
+                    _resultsServer.InitializeResultsClient(new Uri(resultsReceiverEndpoint), liveConsoleFeedUrl, accessToken);
+                    _resultsClientInitiated = true;
                 }
 
-                _resultsServer.InitializeResultsClient(new Uri(resultsReceiverEndpoint), liveConsoleFeedUrl, accessToken);
-                _resultsClientInitiated = true;
+                jobRequest.Variables.TryGetValue("system.github.launch_endpoint", out VariableValue launchEndpointVariable);
+                var launchReceiverEndpoint = launchEndpointVariable?.Value;
+                if (!string.IsNullOrEmpty(launchReceiverEndpoint))
+                {
+                    Trace.Info("Initializing launch client");
+                    _launchServer.InitializeLaunchClient(new Uri(launchReceiverEndpoint), accessToken);
```

Comment:
we are using a JWT token from Token service to auth with Laucn?

Reply:
Yes, that's correct